Metrics Updates || Update Metric with Action Advancement.

erinmikailstaples opened the pull request:
## Describe your changes

Restructuring Action Advancement to new template.

## Shortcut ticket

For Galileo internally raised PRs only, please update this with your shortcut ticket.

[SC-37993](https://app.shortcut.com/galileo/story/37993/metrics-update-action-advancement)

For external PRs, please add the issue (just put the number after the # below, and GitHub will automatically create a link):

## Checklist before requesting a review

- [x] - Is this ready for review? If not, raise as a draft PR
- [x] - This deployed to a staging environment correctly
- [x] - I have reviewed my changes
- [x] - I have reviewed the deployed version of my changes
- [x] - I have tested any code that is added or updated
- [x] - I have verified all images and videos are clear, with appropriate zoom
- [x] - All checks have passed
- [x] - This references a feature that is public. If not, add a note and we can schedule the merge for after the feature release

dmcwhorter commented on concepts/metrics/agentic/action-advancement.mdx:
```diff
@@ -6,109 +6,234 @@ description: Understand how to measure and optimize the effectiveness of your AI
+| **Name of Metric** | Action Advancement |
+| **Metric Category** | Agentic Metrics |
+| **Use this metric for** | Evaluating whether AI agents make progress toward user goals in conversations |
+| **Can be applied to** | agent, session, trace, + span |
```
what do these values align with?
Could we make these align with the supported step types in galileo?  
reference: https://github.com/rungalileo/core/blob/4fbb88dfe7113eb922fb32244f74f850fb18cd0d/src/galileo_core/schemas/logging/step.py#L18-L25
corresponding to:
1. session
2. trace
3. workflow span
3. agent span
3. retriever span
3. llm span
3. tool span

And could we link from each of these to a spot in the docs that explains the logging model and what each type means?

erinmikailstaples replied:
- oooh love this, thank you @dmcwhorter — also thank you for linking where it's at **adds to bookmarks**

dmcwhorter commented on concepts/metrics/agentic/action-advancement.mdx:
```diff
@@ -6,109 +6,234 @@ description: Understand how to measure and optimize the effectiveness of your AI
-
+  When optimizing for Action Advancement, ensure you're not sacrificing other important aspects like safety, factual accuracy, or user experience in pursuit of task completion.
 </Note>
+
```
what about a "Troubleshooting" section here with common problems with the *metric* itself and how to fix them?

vatsalgl commented on concepts/metrics/agentic/action-advancement.mdx:
```diff
@@ -6,109 +6,234 @@ description: Understand how to measure and optimize the effectiveness of your AI
+| **Protect Support** | No - Not applicable for this metric |
+| **Constants** | None - Uses dynamic evaluation |
+| **Usage Context** | Agentic workflows, multi-step tasks, tool-using assistants |
+| **Value Type** | Percentage (0.0 to 1.0) - Binary judgment converted to percentage |
```
We can improve our wording here. There is a common misconception that 60% action_advancement means 60% of the actions have advanced. But it actually means "there is a 60% chance that any 1 action has advanced". The percentages for any boolean metrics are just "confidence scores"

erinmikailstaples commented on concepts/metrics/agentic/action-advancement.mdx:
```diff
@@ -6,109 +6,234 @@ description: Understand how to measure and optimize the effectiveness of your AI
+  </Step>
+  
   <Step title="Score Calculation">
     The final Action Advancement score is computed as the percentage of positive ('yes') responses out of all evaluation responses.
```
updated! Thank you @vatsalgl

dmcwhorter reviewed (approved):
looks great!

erinmikailstaples commented:
Hey all — would love a second set of eyes here on this.

- Reduced redundancies
- Added more detail about explicitness
- Cleaned up layout around cards
- Corrected misconceptions

Currently blocked by: 
- Troubleshooting/details on how to evaluate it.

This PR is good to go, but have opened a second shortcut ticket to update this when troubleshooting is added.
